kubernetes/samples/scripts/create-weblogic-domain/model-in-image/build-model-image.sh

```diff
@@ -0,0 +1,144 @@
+#      for the 'WLS' domain type, and otherwise defaults to 
+#         'container-registry.oracle.com/middleware/fmw-infrastructure'. 
+#      The tag defaults to '12.2.1.4'.
+#
+#    MODEL_IMAGE_NAME, MODEL_IMAGE_TAG:
+#      Defaults to 'model-in-image' and 'v1'.
+#
+
+set -eu
+set -o pipefail
+
+SCRIPTDIR="$( cd "$(dirname "$0")" > /dev/null 2>&1 ; pwd -P )"
+source $SCRIPTDIR/env-init.sh
+
+cd ${WORKDIR}
+
+echo @@ Info: WDT_DOMAIN_TYPE=${WDT_DOMAIN_TYPE}
+echo @@ Info: MODEL_DIR=${MODEL_DIR}
+echo @@ Info: BASE_IMAGE_NAME=${BASE_IMAGE_NAME}
+echo @@ Info: BASE_IMAGE_TAG=${BASE_IMAGE_TAG}
+echo @@ Info: MODEL_IMAGE_NAME=${MODEL_IMAGE_NAME}
+echo @@ Info: MODEL_IMAGE_TAG=${MODEL_IMAGE_TAG}
+echo @@ Info: MODEL_IMAGE_BUILD=${MODEL_IMAGE_BUILD}
+
+if [ ! "$MODEL_IMAGE_BUILD" = "always" ] \
+   && [ ! -z "$(docker images -q $MODEL_IMAGE)" ]; then
+  echo "@@"
+  echo "@@ Info: ----------------------------------------------------------------------------"
+  echo "@@ Info: NOTE!!!                                                                     "
+  echo "@@ Info:   Skipping model image build because '$MODEL_IMAGE' found in docker images. "
+  echo "@@ Info:   To always build the model image, 'export MODEL_IMAGE_BUILD=always'.       "
+  echo "@@ Info: ----------------------------------------------------------------------------"
+  echo "@@"
+  exit 0
+fi
+
+if [ ! -d "$MODEL_DIR" ]; then
+  echo "@@ Error: MODEL_DIR directory not found. Did you remember to stage it first?"
+  exit 1
+fi
+
+MODEL_YAML_FILES="$(ls $MODEL_DIR/*.yaml | xargs | sed 's/ /,/g')"
+MODEL_ARCHIVE_FILES="$(ls $MODEL_DIR/*.zip | xargs | sed 's/ /,/g')"
+MODEL_VARIABLE_FILES="$(ls $MODEL_DIR/*.properties | xargs | sed 's/ /,/g')"
+
+echo @@ Info: MODEL_YAML_FILES=${MODEL_YAML_FILES}
+echo @@ Info: MODEL_ARCHIVE_FILES=${MODEL_ARCHIVE_FILES}
+echo @@ Info: MODEL_VARIABLE_FILES=${MODEL_VARIABLE_FILES}
+
+echo @@
+echo @@ Info: Setting up imagetool and populating its cache with the WDT installer
+echo @@
+
+mkdir -p imagetool/cache
+mkdir -p imagetool/bld
+unzip -o weblogic-image-tool.zip
+
+IMGTOOL=${WORKDIR}/imagetool/bin/imagetool.sh
+
+# The image tool uses the WLSIMG_CACHEDIR and WLSIMG_BLDIR env vars:
+export WLSIMG_CACHEDIR=${WORKDIR}/imagetool/cache
+export WLSIMG_BLDDIR=${WORKDIR}/imagetool/bld
+
+set -x
+
+${IMGTOOL} cache deleteEntry --key wdt_myversion
+${IMGTOOL} cache addInstaller \
+  --type wdt --version myversion --path ${WORKDIR}/weblogic-deploy-tooling.zip
+
+set +x
+
+echo "@@"
+echo "@@ Info: Starting model image build for '$MODEL_IMAGE'"
+echo "@@"
+
+#
+# Run the image tool to create the model image. It will use the WDT binaries
+# in the local image tool cache marked with key 'myversion' (see 'cache' commands above).
+# Note: The "${macro:+text}" syntax expands to "" if 'macro' is empty, and to 'text' if it isn't
+#
+
+# TBD test empty cases, even all three empty
+
+set -x
+
+${IMGTOOL} update \
```

Comment:
On our "stagging" discussion, it seems down here when we call WIT that you can just list the models, properties files, and archives. This script could just take the same variables and be a lot simpler.  Then the doc. can show how to call it for the sample models and archives.

Reply:
The script lets you specify a directory that contains the the models/props/archives.   How could it be simpler than that?  As for the staging, I realize it's a level of indirection, but staging gives the user a chance to customize same without obliterating their work on a different image.

Reply:
Are you saying you want the option to specify specific files for each? I had that option for at one point during development, and I can add that back in, but note that that turned out to have it's own confusion in practice.  It added more 'options' which had a side effect of obscuring the overall intent of the sample.  It also discouraged a best practice of having all files needed for an image gathered in one place...

Reply:
Are this and your other comments resolved now?